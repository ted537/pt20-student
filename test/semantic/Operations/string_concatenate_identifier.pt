using input;
val str1 = "Hello";
val str2 = "World";

var str = str1 | str2;
