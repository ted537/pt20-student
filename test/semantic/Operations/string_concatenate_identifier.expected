   .tFileDescriptor
   .tLiteralInteger
   oEmitValue
   % value emitted 1
   .tFileBind
   .tLiteralAddress
   oEmitDataAddress
   % value emitted 0
   .tStoreInteger
     .tSkipString
     oEmitNullAddress
     % value emitted -32767
     .tStringData
     oEmitValue
     % value emitted 5
     oEmitString
     % value emitted 72
     % value emitted 101
     % value emitted 108
     % value emitted 108
     % value emitted 111
     % value emitted 0
    .tLiteralAddress
    .tFetchChar
    oEmitValue
    % value emitted 15
     .tSkipString
     oEmitNullAddress
     % value emitted -32767
     .tStringData
     oEmitValue
     % value emitted 5
     oEmitString
     % value emitted 87
     % value emitted 111
     % value emitted 114
     % value emitted 108
     % value emitted 100
     % value emitted 0
    .tLiteralAddress
    .tFetchChar
    oEmitValue
    % value emitted 30
   .tInitialValue
      .tLiteralAddress
      .tFetchChar
      oEmitValue
      % value emitted 15
      .tLiteralAddress
      .tFetchChar
      oEmitValue
      % value emitted 30
     .tConcatenate
   .tInitEnd
   .tLiteralAddress
    .tStoreChar
 .tTrapBegin
 .tTrap
 oEmitTrapKind(trHalt)
 % value emitted 0
